crates/l2/utils/eth_client/mod.rs

```diff
@@ -405,12 +404,21 @@ impl EthClient {
                                         .to_owned(),
                                 ))?,
                         );
-                        let string_data = abi_decoded_error_data
-                            .get(68..68 + string_length.as_usize())
-                            .ok_or(EthClientError::Custom(
+
+                        let string_len = if string_length > usize::MAX.into() {
+                            return Err(EthClientError::Custom(
+                                "Failed to convert string_length to usize in estimate_gas"
+                                    .to_owned(),
+                            ));
+                        } else {
+                            string_length.as_usize()
+                        };
```

Comment:
Nit. You can explicit `string_len`'s type and then do `string_length.try_into().ok_or(EthClientError::Custom("Failed to convert string_length to usize in estimate_gas".to_owned()))?`.

Reply:
Done in [c7aa3b7](https://github.com/lambdaclass/ethrex/pull/1466/commits/c7aa3b7938549869411b412f53caf64eac0bbb2e)